ssc/cmod_utilityrate5.cpp

```diff
@@ -1381,6 +1383,47 @@ class cm_utilityrate5 : public compute_module
 
 		bool ec_enabled = true; // per 2/25/16 meeting
 		bool dc_enabled = as_boolean("ur_dc_enable");
+		
+		// should be separate input
+		bool en_ts_buy_rate = as_boolean("ur_en_ts_buy_rate");
+
+		if (en_ts_buy_rate)
+		{
+			if (!is_assigned("ur_ts_buy_rate"))
+			{
+				throw exec_error("utilityrate5", util::format("Time step buy rate enabled but no time step buy rates specified."));
+			}
+			else
+			{ // hourly or sub hourly loads for single year
+				size_t cnt;
+				ssc_number_t* ts_br;
+				ts_br = as_array("ur_ts_buy_rate", &cnt);
+				size_t ts_step_per_hour = cnt / 8760;
+				if (ts_step_per_hour < 1 || ts_step_per_hour > 60 || ts_step_per_hour * 8760 != cnt)
+					throw exec_error("utilityrate5", util::format("number of buy rate records (%d) must be equal to number of gen records (%d) or 8760 for each year", (int)cnt, (int)m_num_rec_yearly));
+
+				// assign timestep values for utility rate calculations
+				size_t idx = 0;
+				ssc_number_t br;
+				br = 0;
+				size_t step_per_hour = m_num_rec_yearly / 8760;
+				//time step rates - fill out to number of generation records per year
+				// handle cases
+				// 1. if no time step rate  s
+				// 2. if time step rate  has 8760 and gen has more records
```

Comment:
In the case where gen has > 8760 records and buy rate has 8760, this would not properly interpolate the subhourly records. For example, if gen had 4 steps per hour, all buy rates after March would be 0. Is this likely to be an issue or is this something the GUI checks for?

Reply:
See lines 1417 through 1430 below. cnt is the number of buy rate records and ts_step_per_hour is the number of timesteps for the but rate records and step_per_hour is the number of timesteps of the yearly gen record. So, for the case where the yearly gen has > 8760 records, say steps_per_hour =4 from your example, and the buy rate has 8760=cnt records. Then, ts_steps_per_hour is 1 and idx only gets incremented once per the ii loop over step_per_hour. Thus, the buy rates would not run out after March. 

Make sense? A little convoluted but works without issue.

Reply:
That makes sense, thanks for the detailed explanation!